rfcs/0087-aarch64-tier1.md

```diff
@@ -0,0 +1,93 @@
+more and more stable, and more devices appear having NixOS on ARM support.
+Moving it to a Tier 1 platform will allow us to block release channels on
+aarch64-related build failures, making it easier and safer for ARM users
+to upgrade their systems, and will help in keeping software versions in
+sync between several architectures due to `x86_64-linux` and `aarch64-linux`
+builds sharing a channel.
+
+`aarch64-linux` will benefit from increased percieved binary cache coverage
+as an additional result of channel bumps waiting for aarch64 builds to finish,
+saving on build times for end users.
+
+## Prior art
+There were prior attempts at the same feat, but they failed due to technical
+limitations of Hydra:
+ - NixOS/nixpkgs@74c4e30 - disabled in 2017 because of memory issues
+ - NixOS/nixpkgs#52534, NixOS/nixpkgs@36a0c13 - re-enabled in 2018 to pre-build important outputs
+ - NixOS/nixpkgs@1bfe8f1 - disabled again due to hydra-evaluator issues
+
+# Detailed design
+[design]: #detailed-design
+
+If this RFC is accepted, `aarch64-linux` builds will be added to stable
+and unstable channels' `tested` aggregate jobs on Hydra, giving them ability
+to block channel advances. Hydra will start building aarch64 packages and run
+aarch64-based tests as part of stable and unstable channels, including them in
+the binary cache, increasing its coverage as a result.
+
+It is possible that the availability of aarch64 builders from Equinix Metal will at times be
+reduced, causing delays in aarch64 build capacity. We will extend the
+nixos-org-configurations implementation of hydra-provisioner to dynamically allocate
+aarch64 builders on AWS during these capacity shortfalls.
+
+# Examples and Interactions
+[examples-and-interactions]: #examples-and-interactions
+
+<!-- This section illustrates the detailed design. This section should clarify all
+confusion the reader has from the previous sections. It is especially important
+to counterbalance the desired terseness of the detailed design; if you feel
+your detailed design is rudely short, consider making this section longer
+instead. -->
+
+In [nixos/release-combined.nix](https://github.com/NixOS/nixpkgs/blob/master/nixos/release-combined.nix)
+`aarch64-linux` will be moved to `supportedSystems`, enabling NixOS tests
+to block channel advances in case of failures.
+
+Merging this RFC should happen simultaneously with the merging of documentation
+around configuring qemu-binfmt as a fallback method for building aarch64 packages on
+x86_64 machines. Additionally, a sub-project that's out-of-scope for this RFC may be
+established to catch build failures (of which sightings were reported) when using
+emulation.
+
+The list of NixOS AMIs on NixOS.org will also be extended to include aarch64 images.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+Some build failures could unneccesarily delay channel advances, delaying critical updates.
+
+# Alternatives
+[alternatives]: #alternatives
+
```

Comment:
I am of the opinion a separate channel should be the first step. Show there are enough resources and commitment to keep it green for half a year to a year. If that is the case, we can upgrade it to Tier 1. I don't think generally there is enough understanding in how much day-to-day effort goes into actually keeping the channels green. At the same time, I have no idea how few breakage there is with aarch64 nowadays.

Reply:
I added your suggestion to the manuscript.

Reply:
The channel already blocks on aarch64-linux since late 2018 for the limited support set. So I guess we're ready to upgrade it to Tier 1 since things were kept green for half a year to a year.

It's all about having the jobs being tried to be built. Not about adding new blockers.

Reply:
Might be worth specifying which channel we are talking about here. If you mean the unstable channel: it mostly works. If you mean the stable channel: It has been blocked for a few days when I opened the channel PR and probably nobody noticed yet.